unmarshal.go

```diff
@@ -0,0 +1,36 @@
+package candlelight
+
+import (
+	"github.com/spf13/viper"
+	"go.opentelemetry.io/otel/propagation"
+)
+
+// Unmarshal helps load tracing components from configuration.
+type Unmarshal struct {
+	// AppName is the name of the application to be traced.
+	AppName string
+
+	// Key is the viper configuration key containing the tracing options.
+	Key string
+}
+
+func (u Unmarshal) New(v *viper.Viper) (*Tracing, error) {
```

Comment:
I thought we generally want applications to unmarshal with viper, rather than each library package? Largely, we're also switching to use `arrange` pkg functions for dealing with configuration - is this compatible with that?

Reply:
I was following the pattern in https://github.com/xmidt-org/themis/blob/main/xhttp/xhttpserver/unmarshal.go but realized that the newer version of viper no longer aligns with the Themis config interface https://github.com/xmidt-org/themis/blob/00676a46dcb1249c9cf9f6cd47f4b977fbaf3ec8/config/unmarshaller.go#L19 😢  (minor version update sneak attack)

As for the arrange portion, I don't think this aligns it yet as the viper setup in Themis needs to be updated.

Reply:
The way themis makes each package deal with its own viper stuff is something that we're not moving forward with, though.  I don't think it's backwards incompatible to have applications handle the viper unmarshalling, especially for new things? Having libraries take in the config as the parameter rather than viper allows for the application to deal with unmarshaling, whether that's with `arrange` or not.

What I know of how using arrange would look in the `main.go` of an application, somewhat simplified:
```go
func main() {
	v, logger := setup()
	app := fx.New(
		arrange.LoggerFunc(logger),
		arrange.ForViper(v),
		fx.Supply(logger),
		fx.Supply(v), // backwards compatible for themis pkgs
		fx.Provide(
			arrange.UnmarshalKey("key", candelight.Config{}),
			candlelight.New,
		),
	)
	...
}
```
This is where `New()` takes in `Config` as a parameter.